File: app/src/main/java/com/example/review/Animator/TextColorAnimator.java
package com.example.review.Animator;

import android.animation.ValueAnimator;
import android.widget.TextView;

public class TextColorAnimator {
    public static ValueAnimator ofArgb(final TextView view, int... values) {
        ValueAnimator valueAnimator = ValueAnimator.ofArgb(values);

        valueAnimator.addUpdateListener(new ValueAnimator.AnimatorUpdateListener() {
            @Override
            public void onAnimationUpdate(ValueAnimator animation) {
                view.setTextColor((int) animation.getAnimatedValue());
            }
        });
        return valueAnimator;
    }
}

File: app/src/main/java/com/example/review/Animator/TextColorAnimator.kt
package com.example.review.Animator

import android.animation.ValueAnimator
import android.widget.TextView

object TextColorAnimator {
    fun ofArgb(view: TextView, vararg values: Int): ValueAnimator {
        val valueAnimator = ValueAnimator.ofArgb(*values)
        valueAnimator.addUpdateListener { animation -> view.setTextColor(animation.animatedValue as Int) }
        return valueAnimator
    }
}